src/pal/src/map/map.cpp

```diff
@@ -2255,25 +2255,13 @@ void * MAPMapPEFile(HANDLE hFile)
     //Step 1: Read the PE headers and reserve enough space for the whole image somewhere.
     IMAGE_DOS_HEADER dosHeader;
     IMAGE_NT_HEADERS ntHeader;
-    errno = 0;
-    if (0 != lseek(fd, 0, SEEK_SET))
-    {
-        palError = FILEGetLastErrorFromErrno();
-        ERROR_(LOADER)( "lseek failed\n" );
-        goto done;
-    }
-    if (sizeof(dosHeader) != read(fd, &dosHeader, sizeof(dosHeader)))
+    if (sizeof(dosHeader) != pread(fd, &dosHeader, sizeof(dosHeader), 0))
```

Comment:
Historically NT only allowed two pages of headers. 96 sections.
And most images only have one page of headers.
esp. .NET images are of typically limited form.
i.e. this can just be one 4k or 8k read, one syscall instead of two.
And then conditionally but essentially never another read.

You could also just mmap the entire thing, if you don't mind doing that for failure cases.

Reply:
Good info; didn't know that. I would have to measure to know if a 8k read with 1 syscall is faster than two pread() calls reading just what's needed, though.

Using mmap() is possible, yes, but it's at least two other syscalls (mmap() and munmap()), so I don't think's a good idea here; unless you're suggesting to mmap the entire file and then mprotect() the regions that need to be read-only, executable, etc?  This might be a good idea for the future.